### PR TITLE
Fix some PR comment

### PR DESCRIPTION
**Pull request by chibatching**
## Issue
- none

## Overview (Required)
- Fix some PR comment from @jmatsu #359
  - 112fd7b : Swap if statement condition orde https://github.com/DroidKaigi/conference-app-2017/pull/359#pullrequestreview-22684993
  - 915a451 : Rename constant value https://github.com/DroidKaigi/conference-app-2017/pull/359#pullrequestreview-22685493
  - 33614bb : Use Rect instead of Point and primitive integer https://github.com/DroidKaigi/conference-app-2017/pull/359#pullrequestreview-22685234
  - 2ad1552 : Simplify AppBarLayout expanded/collapsed check https://github.com/DroidKaigi/conference-app-2017/pull/359#pullrequestreview-22686021
  - fca3292 : Check callback is null or not https://github.com/DroidKaigi/conference-app-2017/pull/359#pullrequestreview-22685520
  - 0d77f42 : Fix duplicate view translation when AppBar is null

## Links
- https://github.com/DroidKaigi/conference-app-2017/pull/359

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java [hunk 1]
```diff
@@ -51,14 +49,16 @@ public OverScrollLayout(Context context, AttributeSet attrs, int defStyleAttr) {
     @Override
     protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
         super.onLayout(changed, left, top, right, bottom);
-        originalLocation.set(left, top);
-        originalHeight = bottom - top;
-        overScrollThreshold = (int) (originalHeight * OVER_SCROLL_THRESHOLD);
+        originalRect.left = left;
```
Comment (jmatsu) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java]: Rect also has `set` method.
[https://developer.android.com/reference/android/graphics/Rect.html#set(int, int, int, int)](https://developer.android.com/reference/android/graphics/Rect.html#set(int, int, int, int))

Can you use `set` instead?

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java [hunk 2]
```diff
@@ -89,29 +89,24 @@ public void onNestedScroll(View target, int dxConsumed, int dyConsumed, int dxUn
             }
         }
 
-        if (appBarLayout == null || scrollTop && dyUnconsumed < 0 && isAppBarExpanded(appBarLayout)) {
-            translationY(-dyUnconsumed);
-        }
-        if (appBarLayout == null || scrollEnd && dyUnconsumed > 0 && isAppBarCollapsed(appBarLayout)) {
+        if (appBarLayout == null
+                || (scrollTop && dyUnconsumed < 0 && isAppBarExpanded(appBarLayout))
```
Comment (jmatsu) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java]: Improved readability. I love this style. 👍

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java [hunk 3]
```diff
@@ -138,11 +132,11 @@ public void onAnimationRepeat(Animator animation) {
                     });
             return;
         }
-        if (getY() != originalLocation.y) {
+        if (getY() != originalRect.top) {
```
Comment (jmatsu) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java]: I have a question. 🙋 
In here's statement, is it needless to check `overScrollListener !=null`?
I think this animation is to restore `overscroll` animation above, so I'm wondering this is really needed if `overScrollListener == null`.

Comment (chibatching) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java]: Oops, it's my mistake to insert null check at first if statement. 😇 
I think it is just need in onAnimationEnd.
IMO, dismissing view and restoring view animation should be performed regardless of whether listener is null or not.
Difference should be just performed callback or not.

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java (hunk 1 above)
Comment (chibatching) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java]: Fix on 624ee8c

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java (hunk 3 above)
Comment (chibatching) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java]: Fix on c32852a

Comment (jmatsu) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java]: I see. Sounds reasonable.

**Comment by jmatsu** [comment]
Mmm, why do tests fail... 🤔 

@chibatching 
So sorry, could you please merge or rebase master?

**Comment by chibatching** [comment]
I run `./gradlew lint` on local machine on this and master branch, and I got some lint error.
I think we should fix this lint error.

**Comment by jmatsu** [comment]
@chibatching 
Got it. Thank you for reporting! We are investigating it...

**Comment by jmatsu** [comment]
@chibatching 
Finally, fixed. 🎉 
It was caused by a connection between circleci and github 😞 

Anyway, your changes LGTM!
Thank you for tackling!

**Comment by chibatching** [comment]
Thanks for review and merge!!

**Pull request merged**